string[] linkGroupNames
string[] linkNames
string[] objectNames
